sdk/python/kfp/cli/compile_.py

```diff
@@ -133,12 +134,19 @@ def parse_parameters(parameters: Optional[str]) -> Dict:
     is_flag=True,
     default=False,
     help='Whether to disable type checking.')
+@click.option(
+    '--enable-caching/--disable-caching',
```

Comment:
Change this to `--disable-default-caching` to maintain precedence with other flags in this file. (like the `type_check` one)